fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/SelectExpression.java

```diff
@@ -408,9 +408,9 @@ public Iterable<MatchInfo> subsumedBy(@Nonnull final RelationalExpression candid
                     .stream()
                     .allMatch(QueryPredicate::isTautology);
             if (allNonFiltering) {
-                return MatchInfo.tryMerge(partialMatchMap, mergedParameterBindingMap, PredicateMap.empty(), remainingValueComputationOptional)
-                        .map(ImmutableList::of)
-                        .orElse(ImmutableList.of());
+                return MatchInfo.tryMerge(partialMatchMap, ImmutableMap.of(), PredicateMap.empty(), PredicateMap.empty(), Optional.empty(), Optional.empty())
```

Comment:
Why is the `remainingValueComputaionOptional` gone?

Reply:
That is definitely a mistake. Reverted.